specs/protocol/derivation.md

```diff
@@ -1278,3 +1285,29 @@ follows:
 
 [extended-attributes]: exec-engine.md#extended-payloadattributesv1
 [Fee Vaults]: exec-engine.md#fee-vaults
+
+## On Future-Proof Transaction Log Derivation
+
+As described in [L1 Retrieval][#l1-retrieval], batcher transactions' types are required to be from a fixed allow-list.
+
+However, we want to allow deposit transactions and `SystemConfig` update events to get derived even from receipts of
+future transaction types, as long as the receipts can be decoded following a best-effort process:
+
+As long as a future transaction type follows the [EIP-2718][https://eips.ethereum.org/EIPS/eip-2718] specification,
+the type can be decoded as the first byte of an encoded transaction or receipt (excluding legacy transactions, which are
+not _future_ transactions). We can then proceed as follows to get the logs of such a future transaction, or discard the
+transaction's receipt as invalid.
+
+- If it's a known transaction type, that is, legacy (first byte of the encoding is in the range `[0xc0, 0xfe]`) or its
+first byte is in the range `[0, 4]` or `0x7e` (_deposited_), then it's not a _future transaction_ and we know how to
+decode the receipt and this process is irrelevant.
+- If a transaction's first byte is in the range `[5, 0x7d]`, it is expected to be a _future_ EIP-2718 transaction, so we
+can proceed to the receipt. Note that we excluded `0x7e` because that's the deposit transaction type, which is known.
+- The _future_ receipt encoding's first byte must be the same byte as the transaction encoding's first byte, or it is
+discarded as invalid, because we require it to be an EIP-2718-encoded receipt to continue.
+- The receipt payload is decoded as if it is encoded as `rlp([status, cumulative_transaction_gas_used, logs_bloom,
+logs])`, which is the encoding of the known non-legacy transaction types.
+  - If this decoding fails, the transaction's receipt is discarded as invalid.
```

Comment:
Would you agree with this summary:

> WIth this spec change, we are saying that any transaction which is both:
> * EIP 2718 compliant
> * has a receipt whose encoding is_also_ compliant with the existing non-legacy RLP encoding scheme
> 
> Is supported by derivation in the sense that deposits and systemconfig events can be derived from such transaction receipts.
> 

So for example if in the coming years we get tx type 8 coming along with the usual RLP receipt encoding, we don't need to change the spec. If we then get tx type 9 with a different receipt structure, we would need to revisit this part of the spec?

Reply:
Yes